### DISCUSSION TITLE
Minio single node on HDD is very slow when putting small files

### DISCUSSION THREAD

**Discussion by kietheros** [minio/minio#18523] [Q&A] [answered]
Hi everyone, I benchmark my minio on HDD, puting small files (100kb)
## 1. Test HDD
```
dd if=/dev/zero of=/mnt/disk1/test1.img bs=100k count=90240  oflag=direct
Throughput: 251 MB/s

dperf -b 100KiB /mnt/disk{1..4} --verbose
┌────────────┬───────────┬───────────┬───┐

│ PATH       │ READ      │ WRITE     │   │

│ /mnt/disk4 │ 256 MiB/s │ 113 MiB/s │ ✓ │

│ /mnt/disk2 │ 117 MiB/s │ 119 MiB/s │ ✓ │

│ /mnt/disk3 │ 96 MiB/s  │ 137 MiB/s │ ✓ │

│ /mnt/disk1 │ 79 MiB/s  │ 190 MiB/s │ ✓ │

└────────────┴───────────┴───────────┴───┘
```

## 2. Test with minio server 1: only 1 node with 1 HDD
Tool: warp, run on localhost, 40 -> 80 threads
```
Put file 100k: average ~ 20.74 MiB/s
Put file 1M  : average ~ 59.83 MiB/s
```
## 3. Test with minio server 2: only 1 node with 4 simliar HDDs, EC:2
Tool: warp, run on localhost, 40 -> 80 threads
```
Put file 100k: average ~ 18.76 MiB/s
Put file 1M  : average ~ 69.92 MiB/s
```
Write throughput of both minio servers are much slower than HDD's. Minio 2 doesn't also improve performance. I think throughput of minio 2 should increase two times !!!!.

I have read the document about minio benchmark on HDD, they setup minio with 16 nodes and put file with size 2G. I don't find any benchmarks with one node like my minio. 
Could anyone review my test ? How can I do to improve minio performance when writing small files ? (only 1 node).

**Comment by harshavardhana** [comment, marked as answer]
HDDs are never meant for IOPs use NVMe.
